tests/main/prepare-image-validation-sets/task.yaml

```diff
@@ -0,0 +1,162 @@
+summary: Verify that we are able to use validation sets with the model assertion through prepare-image 
+
+# autopkgtest run only a subset of tests that deals with the integration
+# with the distro
+backends: [-autopkgtest]
+
+# disable the following distributions
+# ubuntu-14, lack of systemd-run
+# ubuntu-18.04-32: pc gadget from 20 is not available
+# ubuntu-20.04-arm*, because we use pc kernel and gadget.
+systems:
+- -ubuntu-14.04-*
+- -ubuntu-18.04-32
+- -ubuntu-20.04-arm-*
+
+environment:
+    ROOT: /home/test/tmp/
+    IMAGE: /home/test/tmp/system-seed
+    GADGET: /home/test/tmp/gadget
+    STORE_DIR: $(pwd)/fake-store-blobdir
+    STORE_ADDR: localhost:11028
+
+prepare: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+
+    mkdir -p "$ROOT"
+    chown test:test "$ROOT"
+
+restore: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+    
+    "$TESTSTOOLS"/store-state teardown-fake-store "$STORE_DIR"
+    rm -rf "$ROOT"
+
+execute: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+
+    install_snap_to_fakestore() {
+        local SNAP_NAME="$1"
+        local SNAP_REVISION="$2"
+        local SNAP_PATH
+
+        SNAP_PATH=$("$TESTSTOOLS"/snaps-state pack-local "$SNAP_NAME")
+        "$TESTSTOOLS"/store-state make-snap-installable "$STORE_DIR" "$SNAP_PATH"
+        cat > snap-"$SNAP_NAME"-decl.json <<EOF
+    {
+        "type": "snap-declaration",
+        "format": "1",
+        "revision": "1",
+        "snap-name": "$SNAP_NAME",
+        "snap-id": "$SNAP_NAME-id",
+        "plugs": {
+            "snapd-control": {
+            "allow-installation": "true",
+            "allow-auto-connection": "true"
+            }
+        }
+    }
+    EOF
+        cat > snap-"$SNAP_NAME"-rev.json <<EOF
+    {
+        "type": "snap-revision",
+        "snap-id": "$SNAP_NAME-id",
+        "snap-revision": "$SNAP_REVISION"
+    }
+    EOF
+        fakestore new-snap-declaration --dir "${STORE_DIR}" "$SNAP_PATH" --snap-decl-json snap-"$SNAP_NAME"-decl.json
+        fakestore new-snap-revision --dir "${STORE_DIR}" "$SNAP_PATH" --snap-rev-json snap-"$SNAP_NAME"-rev.json
+    }
+    
+    # Ensure the following snaps are available on the host snap to avoid the
+    # store script trying to fetch these after setting up the fake store.
+    echo Installing required snaps we will need
+    snap install core
+    snap install remarshal
```

Comment:
Is install of `remarshal` required? It looks like `... store-state make-snap-installable ...` will take care of this? Also for `jq`

Reply:
It's a good observation, the issue being here that when calling `make-snap-installable` the fake-store is setup, this means that if it tries to download those snaps, it will contact the fake-store instead (and fail, since they are not present).

Reply:
Got it thanks.